core/src/main/resources/org/apache/spark/ui/static/sorttable.js

```diff
@@ -184,6 +184,9 @@ sorttable = {
       if (text != '') {
         if (text.match(/^-?[£$¤]?[\d,.]+%?$/)) {
           return sorttable.sort_numeric;
+        } else if (text.match(/^(\d+(?:\.\d+)?)\s*(b|[k|m|g|t|p|e|z|y]ib)[\s\S]*/i)) {
```

Comment:
I think the following should simply work:
```
text.match(/^(\d+(\.\d+)?)\s*(b|[k|m|g|t|p|e|z|y]ib)/i
```
The `?:` and `[\s\S]*` part is not necessary here.

Reply:
done this.